MapboxNavigation/NavigationMapView.swift

```diff
@@ -1335,3 +1367,122 @@ open class NavigationMapView: MGLMapView, UIGestureRecognizerDelegate {
+        highlightedBuildingsLayer.fillExtrusionHeightTransition = MGLTransition(duration: 0.8, delay: 0)
+        highlightedBuildingsLayer.fillExtrusionOpacityTransition = MGLTransition(duration: 0.8, delay: 0)
+        
+        style?.addLayer(highlightedBuildingsLayer)
+        
+        return highlightedBuildingsLayer
+    }
+
+    private func buildingId(at coordinate: CLLocationCoordinate2D) -> Int64 {
+        let screenCoordinate = convert(coordinate, toPointTo: self)
+
+        // To increase a chance of selecting correct building ID filter out
+        // features which do not contain appropriate attributes.
+        let features = visibleFeatures(at: screenCoordinate).filter {
+            let extrude = $0.attribute(forKey: "extrude") as? String
+            let type = $0.attribute(forKey: "type") as? String
+            return extrude == "true" && type == "building"
+        }
+        
+        if let feature = features.first, let buildingId = feature.identifier as? Int64 {
+            return buildingId
+        }
+        
+        return BuildingHighlightAttributes.kDefaultIdentifier
+    }
+    
+    private func highlightBuildings(_ buildings: [BuildingHighlightAttributes], in3D: Bool = false, extrudeAll: Bool = false) {
+        // In case if array with highlighted building attributes is empty - do nothing.
+        if buildings.isEmpty { return }
+        // Add layer which will be used to highlight buildings if it wasn't added yet.
+        guard let highlightedBuildingsLayer = addBuildingsLayer() else { return }
+        
+        if extrudeAll {
+            highlightedBuildingsLayer.predicate = NSPredicate(format: "extrude = 'true' AND type = 'building' AND underground = 'false'")
+        } else {
+            // Form a predicate to filter out the other buildings from the datasource so only the desired ones are included.
+            highlightedBuildingsLayer.predicate = NSPredicate(format: "extrude = 'true' AND type = 'building' AND underground = 'false' AND $featureIdentifier IN %@", buildings.map { $0.identifier })
+        }
+        
+        // Buildings with IDs will be highlighted with provided color. Rest of the buildings will be highlighted, but kept at a uniform color.
+        let highlightedBuildingHeightExpression = "MGL_MATCH($featureIdentifier, \(buildings.map { "\($0.identifier), \(in3D ? "height" : "0"), " }.joined(separator: ""))\(extrudeAll ? "height" : "0"))"
+        let highlightedBuildingColorExpression = "MGL_MATCH($featureIdentifier, \(buildings.map { "\($0.identifier), %@, " }.joined(separator: ""))%@)"
+        var colorsList = buildings.map { $0.highlightColor }
+        
+        colorsList.append(buildingDefaultColor)
+        
+        let fillExtrusionHeightStops = [0: NSExpression(forConstantValue: 0),
+                                        13: NSExpression(forConstantValue: 0),
+                                        13.25: NSExpression(format: highlightedBuildingHeightExpression)]
```

Comment:
I think the purpose of these stops is to make the buildings appear to grow out of the ground when they first appear. But if somehow the camera stays put at z13.1, the buildings will stay short. Can we accomplish the same effect without the edge case by setting `fillExtrusionHeight` to `TERNARY($zoomLevel > 13.25, %@, 0)` and relying on the duration in `fillExtrusionHeightTransition` to make the buildings’ appearance less sudden?

Reply:
I did some tests with changed `fillExtrusionHeight` and do not see any significant difference. Maybe @d-prukop or @avi-c can comment on this one?